auth-api/src/auth_api/services/affiliation_invitation.py

```diff
@@ -0,0 +1,391 @@
+                                                                               self._model.entity.business_identifier)
+        self._model.token = confirmation_token
+        updated_invitation: AffiliationInvitationModel = self._model.update_invitation_as_retried(user.identifier)
+        entity: EntityModel = updated_invitation.entity
+
+        business = AffiliationInvitation._get_business_details(entity.business_identifier,
+                                                               RestService.get_service_account_token())
+
+        AffiliationInvitation.send_affiliation_invitation(updated_invitation,
+                                                          business['business']['legalName'],
+                                                          f'{invitation_origin}/{context_path}')
+        return AffiliationInvitation(updated_invitation)
+
+    @staticmethod
+    def delete_affiliation_invitation(invitation_id):
+        """Delete the specified affiliation invitation."""
+        invitation: AffiliationInvitationModel = AffiliationInvitationModel.find_invitation_by_id(invitation_id)
+        if invitation is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        if invitation.status == 'ACCEPTED':
+            raise BusinessException(Error.ACTIONED_AFFILIATION_INVITATION, None)
+
+        check_auth(org_id=invitation.from_org_id, one_of_roles=(ADMIN, COORDINATOR, STAFF))
+
+        invitation.delete()
+
+    @staticmethod
+    @user_context
+    def get_invitations_for_from_org(org_id, status=None, **kwargs):
+        """Get affiliation invitations for from org."""
+        user_from_context: UserContext = kwargs['user_context']
+        org_model = OrgModel.find_by_org_id(org_id)
+        if not org_model:
+            return None
+
+        if status:
+            status = InvitationStatus[status]
+
+        # If staff return full list
+        if user_from_context.is_staff():
+            return AffiliationInvitationModel.find_pending_invitations_by_from_org(org_id)
+
+        current_user: UserService = UserService.find_by_jwt_token()
+        current_user_membership: MembershipModel = \
+            MembershipModel.find_membership_by_user_and_org(user_id=current_user.identifier, org_id=org_id)
+
+        # If no active membership return empty array
+        if current_user_membership is None or \
+                current_user_membership.status != Status.ACTIVE.value:
+            return []
+
+        return AffiliationInvitationModel.find_invitations_from_org(org_id=org_id, status=status)
+
+    @staticmethod
+    @user_context
+    def search_invitations_for_from_org(org_id, status=None, **kwargs):
+        """Search affiliation invitations for from org."""
+        invitation_models = AffiliationInvitation.get_invitations_for_from_org(org_id, status, **kwargs)
+        invitations_result = {
+            'affiliationInvitations': []
+        }
+        for invitation in invitation_models:
```

Comment:
Possible to use list comprehension for this?

Reply:
```
invitations_result['affiliationInvitations'] = [AffiliationInvitation(invitation).as_dict() for invitation in invitation_models]

```

Reply:
Done, thanks for the suggestion

Reply:
I used chatgpt for this suggestion - could always just throw some code in there - ask it to optimize